contracts/message/apps/RFQ.sol

```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity >=0.8.9;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "../framework/MessageSenderApp.sol";
+import "../framework/MessageReceiverApp.sol";
+import "../../safeguard/Pauser.sol";
+
+/** @title rfq contract */
+contract RFQ is MessageSenderApp, MessageReceiverApp, Pauser, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    struct Quote {
+        uint64 srcChainId;
+        address srcToken;
+        uint256 srcAmount;
+        uint64 dstChainId;
+        address dstToken;
+        uint256 dstAmount;
+        uint64 deadline;
+        uint64 nonce;
+        address sender;
+        address receiver;
+        address refundTo;
+        address liquidityProvider;
+    }
+
+    mapping(uint64 => address) public remoteRfqContracts;
+    mapping(bytes32 => bool) public unconsumedMsg;
+    // pending quotes on src chain
+    mapping(bytes32 => bool) public quotes;
+    // executed quotes on dst chain
+    mapping(bytes32 => bool) public executedQuotes;
+    address public treasuryAddr;
+    uint32 public feePercGlobal;
+    // dstChainId => feePercOverride
+    mapping(uint64 => uint32) public feePercOverride;
+
+    event SrcDeposited(bytes32 hash, Quote detail, address srcRecipient, uint64 submissionDeadline);
```

Comment:
`srcRecipent` -> `liquidityProvider`
`submissionDeadline` not needed in the event?